cluster/gce/util.sh

```diff
@@ -132,6 +132,26 @@ function detect-project () {
   fi
 }
 
+function sha1sum-file() {
+  if which shasum >/dev/null 2>&1; then
+    shasum -a1 "$1" | awk '{ print $1 }'
+  else
+    sha1sum "$1" | awk '{ print $1 }'
+  fi
+}
+
+function already-staged() {
+  local -r file=$1
+  local -r newsum=$2
+
+  if [ ! -e "${file}.sha1" ]; then
```

Comment:
Please always use `[[ ... ]]` in bash, it's a built-in and allows for more flexible syntax.

In this case, you can use `||` instead of the `if !`.

```shell
  [[ -e "${file}.sha1" ]] || return 1
```